File: C++/Material.cpp
#include "Color.hpp"
#include "Material.hpp"
Material::Material(){}
Material::Material(Color col_value)
         :color{col_value},ambient{0.05}, diffuse{1.0}, specular{1.0}, reflection{0.5}{}

Material::Material(Color col_value, double amb_val, double diff_val, double spec_val, double ref_val)
         :color{col_value}, ambient{amb_val}, diffuse{diff_val}, specular{spec_val}, reflection{ref_val}{}


Chequered_Material::Chequered_Material(Color col1, Color col2)
                   :Material{col1}, color2{col2}{}

Color Material::color_at_surface(Vec3 hit_pos)const
{

    return this->color;
}

Color Chequered_Material::color_at_surface(Vec3 hit_pos)const
{
    if (int((hit_pos.x + 5.0)* 3.0) % 2 == int(hit_pos.z * 3.0)%2)
    {
        return this->color;
    }
    else
    {
        return this->color2;
    }

}

File: Python/Material.py
# -*- coding: utf-8 -*-
"""
Created on Sat Jun  6 22:00:40 2020

@author: Aditya
"""
class material:
    def __init__(self, color = None, ambient = 0.05, diffuse = 1.0, specular = 1.0, reflectivity = 0.5):
        self.color = color
        self.ambient = ambient
        self.diffuse = diffuse
        self.specular = specular
        self.reflectivity = reflectivity
    
    def find_color(self, position):
        return self.color


class material_check(material):
    def __init__(self, color1, color2):
        super().__init__()
        self.color1 = color1;
        self.color2 = color2;
        
    def find_color(self, position):
        # import pdb; pdb.set_trace()
        if int((position[0] + 5.0)*3.0)%2 == int(position[2]*3.0)%2:
            return self.color1
        else:
            return self.color2
